src/base/AsyncJob.cc

```diff
@@ -179,16 +178,15 @@ void AsyncJob::callEnd()
 }
 
 // returns a temporary string depicting transaction status, for debugging
-const char *AsyncJob::status() const
+const char *
+AsyncJob::status() const
 {
     static MemBuf buf;
     buf.reset();
 
-    buf.append(" [", 2);
```

Comment:
A PR changing many (all?) AsyncJob::status() implementations should fix that API to allow for compounded base+kid+gradkid status reports. We probably want to split that API into two because the current status() is used in many debugs() messages where output requirements are rather different (this PR probably breaks some of those use cases).

Reply:
It is possible that it breaks some use cases by missing out on spacing, I've fixed what I could find.
I'm not convinced that it's necessary to have two different functions to report status, in the end the only difference is formatting, isn't it?

Reply:
> the only difference is formatting, isn't it?

No, it is not, not for any reasonable definition of "formatting" in this context.

debugs() status():

* target audience: developers (that can easily decipher cryptic output by reading Squid code)
* should be very compact -- often added to many (already long) debugs() lines
* should be fairly fast -- used often when debugging at level 3+

job state reported inside a mgr:jobs report:

* target audience: admins (humans that do not read Squid code) and their tools
* output size is not a concern; new lines and such are very useful for readability
* output speed is not a significant concern

Reply:
> @yadij requested your review on: Improve YAML compliance of mgr:jobs report.

I am not sure why my re-review was requested before any attempt to address my change request. FWIW, until this change request is addressed, polishing minor PR changes is likely to result in quite a few wasted cycles because at least some of that polishing will need to be redone after this change request is addressed.

If there is interest, I can suggest a way to reshape this PR to avoid modifying existing status() implementations (which triggers status() API change requests and other complications) while still providing YAML-compliant mgr:jobs reports.